lib/board.rb

```diff
@@ -0,0 +1,116 @@
+require './lib/cell'
+require "pry"
+
+class Board
+
+  attr_reader :cells
+
+  def initialize()
+    @cells = {
+      "A1" => Cell.new("A1"),
+      "A2" => Cell.new("A2"),
+      "A3" => Cell.new("A3"),
+      "A4" => Cell.new("A4"),
+      "B1" => Cell.new("B1"),
+      "B2" => Cell.new("B2"),
+      "B3" => Cell.new("B3"),
+      "B4" => Cell.new("B4"),
+      "C1" => Cell.new("C1"),
+      "C2" => Cell.new("C2"),
+      "C3" => Cell.new("C3"),
+      "C4" => Cell.new("C4"),
+      "D1" => Cell.new("D2"),
+      "D2" => Cell.new("D4"),
+      "D3" => Cell.new("D3"),
+      "D4" => Cell.new("D4") }
+  end
+
+  def valid_coordinate?(coords)
+    cells.find do |cell|
+      return true if cell[1].coordinates == coords
+    end
+  end
+
+  def valid_placement?(ship, placements)
+    # if ship overlaps another ship then returns false immediately
+    if ship_overlap?(placements) == false
```

Comment:
Call to new method to check if ship overlaps another ship. If it does, returns false as the rest of the placement is irrelevant. Correct?